core/src/main/java/org/mule/registry/MuleRegistryHelper.java

```diff
@@ -725,30 +712,59 @@ public Object applyProcessorsAndLifecycle(Object object) throws MuleException
     /**
      * {@inheritDoc}
      */
+    @Override
     public Object applyProcessors(Object object) throws MuleException
     {
-        return registry.getTransientRegistry().applyProcessors(object, null);
+        return inject(object);
     }
 
     /**
      * {@inheritDoc}
      */
+    @Override
+    @Deprecated
     public Object applyProcessors(Object object, int flags) throws MuleException
     {
-        return registry.getTransientRegistry().applyProcessors(object, flags);
+        return inject(object);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Object inject(Object object) throws MuleException
+    {
+        LifecycleRegistry lifecycleRegistry = registry.getLifecycleRegistry();
+        if (lifecycleRegistry != null)
+        {
+            return lifecycleRegistry.inject(object);
```

Comment:
if lifecycleRegistry.inject(Object) returns the same object, then remove the return here.